### PR TITLE
More text rendering improvements

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

- Remove Linq usage in FamilyNameCollection ctor
- Use the familyName as part of FontKey
- Cache Skia objects when possible (SKPaint, SKTextStyle, SKTextBlobBuilder) 
- Run the font matching through cached typefaces first to avoid SKFontManager.MatchCharacter
- Throw an exception when the default font family name is null
- Remove redundant UnicodeGeneralCategory
- [SKIA] Resuse paint for stroke and for fill
## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Skia/Avalonia.Skia/PlatformRenderInterface.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace Avalonia.Skia
     /// <summary>
     /// Skia platform render interface.
     /// </summary>
-    internal class PlatformRenderInterface : IPlatformRenderInterface
+    public class PlatformRenderInterface : IPlatformRenderInterface
```
Comment (Gillibald) [inline src/Skia/Avalonia.Skia/PlatformRenderInterface.cs:15]: Revert this

File: tests/Avalonia.Skia.UnitTests/TextLayoutTests.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public void Should_Apply_TextStyleSpan_To_Text_In_Between()
 
                 var layout = new TextLayout(
                     s_multiLineText,
-                    Typeface.Default,
```
Comment (Gillibald) [inline tests/Avalonia.Skia.UnitTests/TextLayoutTests.cs:32]: Revert this

File: src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs [hunk 3]
```diff
@@ -21,7 +20,7 @@ public FamilyNameCollection(string familyNames)
                 throw new ArgumentNullException(nameof(familyNames));
             }
 
-            Names = familyNames.Split(',').Select(x => x.Trim()).ToArray();
+            Names = Array.ConvertAll(familyNames.Split(','), p => p.Trim());
```
Comment (grokys) [inline src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs:23]: Huh! I never knew this method existed! Need to remember this.

File: src/Avalonia.Visuals/Media/Fonts/FontKey.cs [hunk 4]
```diff
@@ -4,20 +4,20 @@ namespace Avalonia.Media.Fonts
 {
     public readonly struct FontKey : IEquatable<FontKey>
     {
-        public readonly FontFamily FontFamily;
+        public readonly string FamilyName;
```
Comment (grokys) [inline src/Avalonia.Visuals/Media/Fonts/FontKey.cs:7]: Only just noticed this, but is it a good idea to expose fields on a public API? Should these be properties?

**Review by grokys — APPROVED** [review]
Just one question, but LGTM in general.

File: src/Avalonia.Visuals/Media/Fonts/FontKey.cs (hunk 4 above)
Comment (Gillibald) [reply src/Avalonia.Visuals/Media/Fonts/FontKey.cs:7]: Hmm. Good catch. Will make these fields private and expose values via properties.

**Review by MarchingCube — APPROVED** [review]
Checked it out locally as well, nice to see that we also do not allocate so many `SKPaint` objects anymore.

**Comment by Gillibald** [comment]
Y the caching improved the performance a lot. At least I noticed it in the ControlCatalog.

**Pull request merged**